Document grid import / export and energy dashboard use for Enphase Envoy. (#31491)
home-assistant/home-assistant.github.io: source/_integrations/enphase_envoy.markdown

- Historical energy production & consumption
- Power production per-inverter

_Consumption sensors require your Envoy to be properly configured with consumption CT sensors installed._

For Envoy S Metered / IQ Gateway Metered with installed and configured current transformers (CT), additional features are available:

> - Production and consumption sensors for each phase, if CT are installed on more than 1 phase.

Reviewer: ```suggestion
- Production and consumption sensors for each phase, if <abbr title="current transformers">CT</abbr> are installed on more than 1 phase.
```

> - Sensors for net production (grid export) and net consumption (grid import) if the consumption CT is a net-consumption CT.

Reviewer: ```suggestion
- Sensors for net production (grid export) and net consumption (grid import) if the consumption <abbr title="current transformers">CT</abbr> is a net-consumption <abbr title="current transformers">CT</abbr>.
```

- Disabled sensors for:
  
  - Phase net production and net consumption.

>   - Frequency Net consumption CT (Aggregate and phase).

Reviewer: ```suggestion
  - Frequency net consumption <abbr title="current transformers">CT</abbr> (aggregate and phase).
```

>   - Voltage Net consumption CT (Aggregate and phase).[^1]

Reviewer: ```suggestion
  - Voltage net consumption <abbr title="current transformers">CT</abbr> (aggregate and phase).[^1]
```

>   - Metering Status for net consumption and production CT (`normal` | `not-metering` | `check-wiring`) (Aggregate and phase).

Reviewer: ```suggestion
  - Metering status for net consumption and production <abbr title="current transformers">CT</abbr> (`normal` | `not-metering` | `check-wiring`) (Agaregate and phase).
```

>   - Meter status flags active for net consumption and production CT (Aggregate and phase).[^2]

Reviewer: ```suggestion
  - Meter status flags active for net consumption and production <abbr title="current transformers">CT</abbr> (aggregate and phase).[^2]
```

> [^1]: For multi-phase system the Envoy sums the voltages of the phases. May be valid for split-phase, but for 3 phase systems use the individual phases rather then the summed value.

Reviewer: ```suggestion
[^1]: For multi-phase systems, the Envoy sums the voltages of the phases. May be valid for split-phase, but for 3-phase systems use the individual phases rather than the summed value.
```

> [^2]: If this value is non-zero consult the diagnostic report of the envoy and look for `raw_data` - `/ivp/meters` - `statusFlags` for set flags (`production-imbalance` | `negative-production` | `power-on-unused-phase` | `negative-total-consumption`).

Reviewer: ```suggestion
[^2]: If this value is non-zero, consult the diagnostic report of the Envoy and look for `raw_data` - `/ivp/meters` - `statusFlags` for set flags (`production-imbalance` | `negative-production` | `power-on-unused-phase` | `negative-total-consumption`).
```

## Credentials update

This integration supports updating configuration by re-adding the integration and specifying the same or new IP address, username, and password. Use this method if your Enlighten credentials or the device's IP address has changed and needs to be updated.

> ## Energy Dashboard

Reviewer: ```suggestion
## Energy dashboard
```

This integration provides several values suitable for the energy dashboard:

> - For `Solar production` use the `Envoy Lifetime energy production` entity.

Reviewer: ```suggestion
- For `Solar production`, use the `Envoy Lifetime energy production` entity.
```

> - For `Grid consumption` use the `Envoy Lifetime net energy consumption` entity.[^3]

Reviewer: ```suggestion
- For `Grid consumption`, use the `Envoy Lifetime net energy consumption` entity.[^3]
```

> - For `Return to grid` use the `Envoy Lifetime net energy production` entity.[^3]

Reviewer: ```suggestion
- For `Return to grid`, use the `Envoy Lifetime net energy production` entity.[^3]
```

> [^3]: Only when using  Envoy S Metered / IQ Gateway Metered with installed and configured current transformers (CT)

Reviewer: ```suggestion
[^3]: Only applies when using  Envoy S Metered / IQ Gateway Metered with installed and configured current transformers (<abbr title="current transformers">CT</abbr>).
```

> There are no readily available battery energy sensors for use with the `Home Battery storage`. One can consider using the Encharge  `real_power_mw` entity as an input to Riemann integrators for charge (negative) or discharge (positive) values. As the [polling interval](#polling-interval) is 1 minute these may be off though.

Reviewer: ```suggestion
There are no readily available battery energy sensors for use with the `Home Battery storage`. You can consider using the Encharge  `real_power_mw` entity as an input to Riemann integrators for charge (negative) or discharge (positive) values. As the [polling interval](#polling-interval) is 1 minute, these may be off though.
```
